src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -800,15 +802,18 @@ export class SavedObjectsRepository {
 
     const esOptions = {
       // If `pit` is provided, we drop the `index`, otherwise ES returns 400.
-      ...(pit ? {} : { index: this.getIndicesForTypes(allowedTypes) }),
+      index: pit ? undefined : this.getIndicesForTypes(allowedTypes),
       // If `searchAfter` is provided, we drop `from` as it will not be used for pagination.
-      ...(searchAfter ? {} : { from: perPage * (page - 1) }),
+      from: searchAfter ? undefined : perPage * (page - 1),
```

Comment:
@lukeelmers Same question here.

Reply:
Same note as above -- it should be fine to include the key with `undefined`. 

(The SO export functional tests would tell us if anything broke here, as ES returns a 400 if you attempt to provide an `index` with a `pit`)